src/debugAdapter/goDebug.ts

```diff
@@ -503,11 +503,21 @@ class GoDebugSession extends DebugSession {
 		return path.replace(this.delve.program, this.delve.remotePath).split(this.localPathSeparator).join(this.remotePathSeparator);
 	}
 
-	protected toLocalPath(path: string): string {
+	protected toLocalPath(pathToConvert: string): string {
 		if (this.delve.remotePath.length === 0) {
-			return this.convertDebuggerPathToClient(path);
+			return this.convertDebuggerPathToClient(pathToConvert);
 		}
-		return path.replace(this.delve.remotePath, this.delve.program).split(this.remotePathSeparator).join(this.localPathSeparator);
+
+		// Fix for https://github.com/Microsoft/vscode-go/issues/1178
+		// When the pathToConvert is under GOROOT, replace the remote GOROOT with local GOROOT
+		if (!pathToConvert.startsWith(this.delve.remotePath)) {
+			let index = pathToConvert.indexOf('src');
+			let goroot = process.env['GOROOT'];
+			if (goroot) {
+				return path.join(goroot, ...pathToConvert.substr(index).split(this.remotePathSeparator));
+			}
+		}
+		return pathToConvert.replace(this.delve.remotePath, this.delve.program).split(this.remotePathSeparator).join(this.localPathSeparator);
```

Comment:
You use `path.join` above and `.join(this.localPathSeparator);` here, should it be the same?

Reply:
path.join uses the path separator for the local machine so it would be the same `localPathSeparator`
additionally it also takes care of trailing `/` at the end of goroot if there is any.

So if I do a split on the goroot, remove trailing `/` if it exists, then I can use `.join(this.localPathSeparator)`